src/docs/testing/integration-tests/index.md

```diff
@@ -0,0 +1,285 @@
+
+```yaml
+integration_test: ^0.9.0
+flutter_test:
+  sdk: flutter
+flutter_driver:
+  sdk: flutter
+```
+
+Create a new directory `integration_test/` with a new file,
+`<name>_test.dart`:
+
+```dart
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  testWidgets("failing test example", (WidgetTester tester) async {
+    expect(2 + 2, equals(5));
+  });
+}
+```
+
+{{site.alert.note}}
+Note: You should only use `testWidgets` to declare your tests, or errors will
+not be reported correctly.
+{{site.alert.end}}
+
+
+To run tests with `flutter drive`, create a new directory containing a new file,
+`test_driver/integration_test_driver.dart`:
+
+```dart
+import 'package:integration_test/integration_test_driver.dart';
+
+Future<void> main() => integrationDriver();
+```
+
+### Directory Structure
+
+```
+lib/
+  ...
+integration_test/
+  foo_test.dart
+  bar_test.dart
+test/
+  # Other unit tests go here.
+test_driver/
+  integration_test.dart
+```
+
+See also:
+
+ * [integration_test usage][]
+
+## Running Using Flutter Driver
+
+These tests can be launched with the `flutter drive` command, where
+`<DEVICE_ID>`: is the optional device ID or pattern displayed in the output of the
+`flutter devices` command:
+
+```bash
+flutter drive \
+  --driver=test_driver/integration_test.dart \
+  --target=integration_test/foo_test.dart \
+  -d <DEVICE_ID>
+```
+
+This runs the tests in `foo_test.dart` via the adapter in
+`test_driver/integration_test.dart`.
+
+
+### Running in a browser 
+
+First, [Download and install ChromeDriver][] and run it on port 4444:
+
+
+```
+chromedriver --port=4444
+```
+
+In a separate process, run `flutter_drive`:
+
+```
+flutter drive \
+  --driver=test_driver/integration_test_driver.dart \
+  --target=integration_test/counter_test.dart \
+  -d web-server
+```
+
+## Testing on Firebase TestLab 
+
+
+### Android Setup
+Follow the instructions in the [Android Device Testing][] section of the README.
+
+### iOS Setup
+Follow the instructions in the [iOS Device Testing][] section of the README.
+
+### TestLab Project Setup
+
+Go to the [Firebase Console], and create a new project if you don't have one
+already. Then navigate to Quality > Test Lab:
+
+{% asset integration-test/test-lab-1.png class="mw-100" alt="Firebase Test Lab Console" %}
+
+### Uploading an Android APK
+
+Create an APK using Gradle:
+
+```bash
+pushd android
+# flutter build generates files in android/ for building the app
+flutter build apk
+./gradlew app:assembleAndroidTest
+./gradlew app:assembleDebug -Ptarget=integration_test/<name>_test.dart
+popd
+```
+
+Where `<name>_test.dart` is the file created in the Project Setup section.
+
+Drag the "debug" APK from
+`<flutter_project_directory>/build/app/outputs/apk/debug` into the "Android Robo
+Test" target on the web page. This will start a Robo test and allow you to run
```

Comment:
Bold the menu item.   "This will start" ->> "This starts"  In fact, search for "will" in your doc. You can probably remove most of them.   https://developers.google.com/style/tense?hl=en